source/importers/SynchronizeColorsTokens.ts

```diff
@@ -0,0 +1,218 @@
+const fsT = require('fs');
+const axios = require("axios");
+const tokensFilesConfig = require('./configs/SynchronizeColorsTokensConfig.json');
+
+interface IResultFileJsonObject {
+	id: string;
+	label: string;
+	value: string;
+}
+
+interface IResultJsonObject  extends IResultFileJsonObject {
+	weight: number;
+}
+
+interface configFileObject {
+	destination: string,
+	destinationJson: string
+}
+
+interface ITokenJsonObject {
+	id: string;
+	label: string;
+	value: string;
+}
+
+const ImportColorsRaw = (
+	binValues: configFileObject,
+	jsonColors: Array<any>
+) => {
+	let hexToCssVariable: Object = {};
+	try {
+		let result: Array<string> = [];
+		let temporaryColorsJson = {};
+		let resultColorsJson = {};
+
+		result.push(':root {');
+
+		jsonColors.forEach(obj => {
+			const patternColorsToProcess = /RAW\/|theme/i;
+			const patternColorsToIgnore = /deprecated|Pattern/i;
+			const patternTheme = /theme/i;
+			if (obj.name.match(patternColorsToProcess) && obj.name.match(patternColorsToIgnore) === null) {
+				const nameSplit = obj.name.split('/');
+				let colorName = (nameSplit[2] === undefined) ? nameSplit[1] : nameSplit[2];
+				console.log(colorName);
+				obj.values.hex = obj.values.hex.replace(/FFFFFF/gi, function(){
+					return 'fff';
+				});
+
+				const colorFinalName = obj.name.match(patternTheme) ? '--'  + colorName : '--raw-' + colorName;
+				console.log(colorFinalName);
+
+				if (hexToCssVariable[obj.values.hex] === undefined) {
+					hexToCssVariable[obj.values.hex] = colorFinalName ;
+				}
+
+				result.push(colorFinalName + ': ' + obj.values.hex + ';');
+				result.push(colorFinalName + '-rgb' + ': ' + hexToRgb(obj.values.hex) + ';');
+
+				/** JSON object structure */
+				colorName = colorName.replace(/--/i, '').replace(/raw-/i, '');
+				const colorNameSplitted = colorName.split('-');
+				const category = (colorNameSplitted[1] === undefined) ? 'default' : colorNameSplitted[0];
+				const resultJsonObject: IResultJsonObject = {
+					id: binValues.destination + '_' + colorName,
+					label: colorName,
+					value: obj.values.hex,
+					weight: parseInt(colorNameSplitted[1])
+				};
+
+				if (temporaryColorsJson[category] === undefined) {
+					temporaryColorsJson[category] = [];
+				}
+				temporaryColorsJson[category].push(resultJsonObject);
+				temporaryColorsJson[category].sort((a,b) => a.weight - b.weight);
```

Comment:
you're sorting on every color push. For performance reaseons it's better to do that once, after you finish the loop